Formula/npiet.rb

```diff
@@ -0,0 +1,48 @@
+class Npiet < Formula
+  desc "Interpreter for the esoteric, picture-based Piet programming language"
+  homepage "https://www.bertnase.de/npiet"
+  url "https://www.bertnase.de/npiet/npiet-1.3e.tar.gz"
+  sha256 "e819c766d12e1dfaf71561429486ed85b6ae4438da6e7ee06871ab5ce58231c5"
+
+  depends_on "gd"
+  depends_on "libpng"
+  depends_on "homebrew/versions/giflib5"
+
+  def install
+    inreplace "npiet-foogol.c", "<malloc.h>", "<malloc/malloc.h>"
+    inreplace "npiet-foogol.y", "<malloc.h>", "<malloc/malloc.h>"
+
+    system "./configure", "--prefix=#{prefix}"
+    # --mandir doesn't work here, despite --help claiming it does.
+    inreplace "Makefile", "man/man1", "share/man/man1"
+
+    system "make"
+    system "make", "install"
+
+    doc.install "foo" # not a placeholder
+    doc.install "examples"
+
+    rm [examples/"Makefile", examples/"runtest.sh"]
+  end
+
+  test do
+    IO.popen("#{bin}/npiet", :err => [:child, :out]) do |process|
+      output = process.read.split("\n").first
+      assert_match /with GD.*?with GIF.*?with PNG/, output
+    end
+
+    png_output = shell_output("#{bin}/npiet #{examples/"hi.png"}")
+    ppm_output = shell_output("#{bin}/npiet #{examples/"hi.ppm"}")
+    gif_output = shell_output("#{bin}/npiet #{examples/"loop.gif"}")
+
+    assert_match png_output, "Hi\n"
```

Comment:
Will do. I wanted it to line up without having large gaps for the PNG and PPM outputs compared to GIF, but it's not strictly necessary.